File: IT/Variabili, operatori, strutture di controllo/Esercizi/C++/esercizio4.cpp
/* file esercizio4.cpp */

#include <iostream>

using std::cout;
using std::endl;

int main()
{
	int a = 5;
	int b = 10;

	cout << "a = " << a << ", b = " << b << endl;

	int s = a + b;

	cout << a << " + " << b << " = " << s << endl;

	int d = a - b;

	if (d < 0) {
		cout << "Differenza negativa!" << endl;
	}

	cout << a << " - " << b << " = " << d << endl;

	if (d != b - a) {
		cout << "Sottrazione non commutativa. Scambio gli operandi." << endl;
		d = b - a;
		cout << b << " - " << a << " = " << d << endl;
	}

	int p = a * b;

	cout << a << " * " << b << " = " << p << endl;

	double q = (double)a / b;

	cout << a << " / " << b << " = " << q << endl;

	if (q != (double)b / a) {
		cout << "Divisione non commutativa. Scambio gli operandi." << endl;
		q = (double)b / a;
		cout << b << " / " << a << " = " << q << endl;
	}
}

File: IT/Variabili, operatori, strutture di controllo/Esercizi/Python/esercizio4.py
# file esercizio4.py

from __future__ import print_function, division

a = 5
b = 10

print("a = {}, b = {}".format(a, b))

s = a + b

print("{} + {} = {}".format(a, b, s))

d = a - b

if d < 0:
	print("Differenza negativa!")

print("{} - {} = {}".format(a, b, d))

if d is not b-a:
	print("Sottrazione non commutativa. Scambio gli operandi.")
	d = b - a
	print("{} - {} = {}".format(b, a, d))

p = a * b

print("{} * {} = {}".format(a, b, p))

q = a / b

print("{} / {} = {}".format(a, b, q))

if q is not b/a:
	print("Divisione non commutativa. Scambio gli operandi.")
	q = b / a
	print("{} / {} = {}".format(b, a, q))

